### PR TITLE
Issue 13942 decreate version for ok http

### PR DESCRIPTION
**Pull request by chenrujun**

### CODE REVIEW

File: eng/versioning/external_dependencies.txt [hunk 1]
```diff
@@ -29,7 +29,7 @@ com.microsoft.rest.v2:client-runtime;2.1.1
 com.microsoft.sqlserver:mssql-jdbc;7.4.1.jre8
 com.microsoft.azure:spring-data-cosmosdb;2.3.0
 com.microsoft.spring.data.gremlin:spring-data-gremlin;2.3.0
-com.squareup.okhttp3:okhttp;4.8.1
+com.squareup.okhttp3:okhttp;3.14.9
```
Comment (alzimmermsft) [inline eng/versioning/external_dependencies.txt:32]: We shouldn't revert dependency versions a major version, if we need to add a second version of OkHttp for 3.x let's add a new custom tag for it.

Comment (chenrujun) [reply eng/versioning/external_dependencies.txt:32]: Thank you for your check.

I revert the version to make the version consistent with `spring-boot-dependencies`.
If our dependency's version not equal to the `spring-boot-dependencies` managed one, 
our customer may have problem like `NoSuchMethodError`

Refs:
1. Issue: https://github.com/Azure/azure-sdk-for-java/issues/13942
1. `spring-boot-dependencies` bom: https://repo.maven.apache.org/maven2/org/springframework/boot/spring-boot-dependencies/2.3.3.RELEASE/spring-boot-dependencies-2.3.3.RELEASE.pom

Comment (chenrujun) [reply eng/versioning/external_dependencies.txt:32]: After discuss with @jialindai , I'll close this PR.

Comment (chenrujun) [reply eng/versioning/external_dependencies.txt:32]: After discussed with @jialindai , we this it's not good to revert major version. I'll close this PR.

**Comment by jialindai** [comment]
The purpose of this PR is to address the jar version problem:
- sdk depends on okhttp 4.8.1
- spring framework depends on okhttp 3.14.9

After discuss with @chenrujun, we agree that downgrade okhttp version is not an ideal solution here, so close this PR.

**Pull request closed without merging**